### PR TITLE
Replace typestyle with jss & add button test case

### PR DESCRIPTION
**Pull request by prateekshourya29**
# PR Details

## Description

**Add 'react-jss' dependency**
**Update 'jest' dependency**
**Add 'jest-dom' dependency**
**Replace typestyle with jss in theme, breadcrumb, button, helpers/index**
**Move button style to new file**
**Add test case for button and base styles**

## Related Issue

#39 
#40 

## Motivation and Context

Because typestyle doesn't support style testing.

## How Has This Been Tested

Tested Button using 'jest' & 'jest-dom'

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: __tests__/components/Button/Button.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+    expect(button).toHaveStyle(`
+      color: rgb(85, 85, 85);
+      box-shadow: none;
+      border: 1px solid #555555;
+      background: transparent;
+    `);
+  });
+
+  it('should not onClick when loading', () => {
```
Comment (anitnilay20) [inline __tests__/components/Button/Button.spec.tsx:230]:
```suggestion
  it('should not allow click while loading', () => {
```

File: __tests__/components/Button/Button.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    act(() => {
+      ReactDOM.render(<Button component={<p />} />, container);
+    });
+
+    const button = container.querySelector('p');
+
+    expect(button).not.toBeNull();
+  });
+
+  it('should not onClick when disabled', () => {
```
Comment (anitnilay20) [inline __tests__/components/Button/Button.spec.tsx:257]:
```suggestion
  it('should not allow click if disabled', () => {
```

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -37,11 +38,13 @@ export const Button: React.FunctionComponent<ButtonProps> = (props) => {
...
-    return (
-      <ThemeConsumer>
-        {(theme) => {
-          const css = style(props, theme);
+  const theme = useTheme();
+  const css = style({ props, theme });
+
+  return (
+    <ThemeConsumer>
+      {(theme) => {
```
Comment (anitnilay20) [inline src/components/Button/Button.tsx:46]: We don't need `ThemeConsumer` as we are getting theme from `useTheme`.

File: tsconfig.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
   },
   "include": [
     "./src"
-  ],
+, "__tests__/components/Button/helpers/helper.specs.tsx"  ],
```
Comment (anitnilay20) [inline tsconfig.json:28]: Why?

**Review by anitnilay20 — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 4 above)
Comment (prateekshourya29) [reply tsconfig.json]: Resolved in 8fd92b4e11cfc7f5bbfd723370608912e53f3890

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (prateekshourya29) [reply src/components/Button/Button.tsx]: Resolved in 3522fcf824ee04e717c79a860bb0f9f0d11b8a4c

File: __tests__/components/Button/Button.spec.tsx (hunk 1 above)
Comment (prateekshourya29) [reply __tests__/components/Button/Button.spec.tsx]: Resolved in 2dc849680c3d8db7d149e9eddfc74994ee0b93c8

File: __tests__/components/Button/Button.spec.tsx (hunk 2 above)
Comment (prateekshourya29) [reply __tests__/components/Button/Button.spec.tsx]: Resolved in 2dc849680c3d8db7d149e9eddfc74994ee0b93c8

**Review by anitnilay20 — APPROVED** [review]

**Pull request merged**